lib/connections/tcp_dial.go

```diff
@@ -60,7 +60,7 @@ func (tcpDialer) Priority() int {
 }
 
 func (d *tcpDialer) RedialFrequency() time.Duration {
-	return time.Duration(d.cfg.Options().ReconnectIntervalS) * time.Second
+	return time.Duration(d.cfg.Options.ReconnectIntervalS) * time.Second
```

Comment:
We need to pass the actual config to things in CommitConfiguration, because the wrapper still refers to the old config. That affects the listeners, I changed the dialers for symmetry.